lib/Elastica/Reindex.php

```diff
@@ -184,6 +194,23 @@ public function setScript(Script $script)
         $this->setParam(self::SCRIPT, $script);
     }
 
+    public function setQuery(AbstractQuery $query): void
+    {
+        $this->setParam(self::QUERY, $query);
+    }
+
+    public function setPipeline(Pipeline $pipeline): void
+    {
+        $this->setParam(self::PIPELINE, $pipeline);
+    }
+
+    public function setRefresh($value): void
```

Comment:
Should we force the value type to be `bool $value`?

Reply:
The valid values are: `true`, `false` and `wait_for`.
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html#docs-reindex-api-query-params

We could also create some class constants for them.

Reply:
I like your solution, it would be cleaner IMHO and would allow us to validate the parameter... or should it be something for the lower-level client library? :thinking: 

@ruflin , what do you think?

Reply:
In general I like the idea of validation at the same time it can put a burden on "us". Class constant would definitively be helpful but i would not validate that only these are used. If a new param is introduced in ES, we need to add it to support it which I don't think it optimal. Happy to also only have the most common / used ones as constants (or none to get started).

Reply:
I quite agree with the "burden", I just wanted to have a more strict typing here.
IIRC Elasticsearch accepts `"true"` (as string) and `true` (as boolean) as equals. if that's the case: we can allow `setRefresh(string $value)`  and define 3 public constants as "true" "false" and "wait_for".

Reply:
I added some constants via fb3e933ec6d068c0a711f9321ae587593a439ab8

Reply:
Thank you for the new fixes, what about only allowing `string` as parameter here?

Reply:
see #1758